conans/client/generators/json_generator.py

```diff
@@ -0,0 +1,47 @@
+import json
+
+from conans.model import Generator
+
+
+def serialize_cpp_info(cpp_info):
+    keys = [
+        "version",
+        "description",
+        "rootpath",
+        "sysroot",
+        "includedirs", "libdirs", "bindirs", "builddirs", "resdirs",
+        "libs",
+        "defines", "cflags", "cppflags", "sharedlinkflags", "exelinkflags",
+    ]
+    res = {}
+    for key in keys:
+        res[key] = getattr(cpp_info, key)
+    return res
+
+
+class JsonGenerator(Generator):
+    @property
+    def filename(self):
+        return "conaninfo.json"
```

Comment:
*conaninfo.json* might be confusing as it is similar to *conaninfo.txt* that gathers the information of a generated pacakge, not info about package dependencies. I suggest using the convention *conanbuildinfo.json*